### PR TITLE
[EPIC-1517] Refactor the Public GitHub Examples of RTU UI components

### PR DESCRIPTION
**Pull request by Superkachenya**

### CODE REVIEW

File: ReadyToUseUIDemo/Usecases/Data Detect Usecases/UsecaseScanOneDimensionalBarcode.swift [hunk 1]
```diff
@@ -31,14 +32,14 @@ class UsecaseScanOneDimensionalBarcode: Usecase, SBSDKUIBarcodeScannerViewContro
...
-        
-        guard let code = barcodeResults.first else { return }
-        let message = code.rawTextString
-        let title = code.type == SBSDKBarcodeTypeQRCode ? "QR code detected" : "Barcode detected"
-        
+                
         viewController.isRecognitionEnabled = false
-        UIAlertController.showInfoAlert(title, message: message, presenter: viewController) {
-            viewController.isRecognitionEnabled = true
+        viewController.dismiss(animated: true) {
```
Comment (sebastato) [inline ReadyToUseUIDemo/Usecases/Data Detect Usecases/UsecaseScanOneDimensionalBarcode.swift:37]: viewController.presentingViewController?.dismiss(...)

File: ReadyToUseUIDemo/Usecases/Data Detect Usecases/UsecaseScanBarcodeBatch.swift [hunk 2]
```diff
@@ -40,6 +41,13 @@ class UsecaseScanBarcodeBatch: Usecase, SBSDKUIBarcodesBatchScannerViewControlle
     
     func barcodesBatchScannerViewController(_ viewController: SBSDKUIBarcodesBatchScannerViewController,
                                             didFinishWith barcodeResults: [SBSDKUIBarcodeMappedResult]) {
+        viewController.dismiss(animated: true) {
```
Comment (sebastato) [inline ReadyToUseUIDemo/Usecases/Data Detect Usecases/UsecaseScanBarcodeBatch.swift:44]: viewController.presentingViewController?.dismiss(...)

File: ReadyToUseUIDemo/Usecases/UsecaseStartWorkflow.swift [hunk 3]
```diff
@@ -82,6 +82,11 @@ extension UsecaseStartWorkflow: SBSDKUIWorkflowScannerViewControllerDelegate {
     func workflowScanViewController(_ viewController: SBSDKUIWorkflowScannerViewController,
                                     didFinish workflow: SBSDKUIWorkflow,
                                     with results: [SBSDKUIWorkflowStepResult]) {
-        self.showWorkflowResults(results, on: viewController)
+        if let navigationController = self.presenter as? UINavigationController, !results.isEmpty {
+            viewController.dismiss(animated: true) {
```
Comment (sebastato) [inline ReadyToUseUIDemo/Usecases/UsecaseStartWorkflow.swift:86]: viewController.presentingViewController?.dismiss(...)

File: ReadyToUseUIDemo/Generic Document results/GenericDocumentResultListViewController.swift [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    }
+    
+    @objc private func didTapCopy(_ sender: UIBarButtonItem) {
+        UIPasteboard.general.string = self.resultsText
+        let alert = UIAlertController(title: "Data fields copied to clipboard",
+                                      message: nil,
+                                      preferredStyle: .alert)
+        self.present(alert, animated: true, completion: nil)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            alert.dismiss(animated: true, completion: nil)
```
Comment (sebastato) [inline ReadyToUseUIDemo/Generic Document results/GenericDocumentResultListViewController.swift:74]: alert.presentingViewController?.dismiss(...)

**Review by sebastato — CHANGES_REQUESTED** [review]

File: ReadyToUseUIDemo/Usecases/Data Detect Usecases/UsecaseScanTwoDimensionalBarcode.swift [hunk 5]
```diff
@@ -29,18 +30,15 @@ class UsecaseScanTwoDimensionalBarcode: Usecase, SBSDKUIBarcodeScannerViewContro
...
-        }
-        
-        let message = code.rawTextString
-        let title = code.type == SBSDKBarcodeTypeQRCode ? "QR code detected" : "Barcode detected"
-        
         viewController.isRecognitionEnabled = false
-        UIAlertController.showInfoAlert(title, message: message, presenter: viewController) {
-            viewController.isRecognitionEnabled = true
+        
+        viewController.dismiss(animated: true) {
```
[the remark this answers is not in the document]
Comment (Superkachenya) [reply ReadyToUseUIDemo/Usecases/Data Detect Usecases/UsecaseScanTwoDimensionalBarcode.swift:35]: This is what I want to do. Dismiss the `SBSDKUIBarcodeScannerViewController`.

**Review by sebastato — APPROVED** [review]

**Pull request merged**